### PR TITLE
Add without id support for SQLite

### PR DESCRIPTION
**Pull request by mashuai**
SQLite supports `WITHOUT ROWID` at the end of `CREATE TABLE` [SQLite Create Table](https://sqlite.org/lang_createtable.html)

### CODE REVIEW

File: src/ast/mod.rs [hunk 1]
```diff
@@ -687,6 +689,11 @@ impl fmt::Display for Statement {
                 if let Some(query) = query {
                     write!(f, " AS {}", query)?;
                 }
+                if let Some(without_rowid) = *without_rowid {
+                    if without_rowid {
+                        write!(f, "WITHOUT ROWID")?;
```
Comment (Dandandan) [inline src/ast/mod.rs:694]: I think we should add a space before `WITHOUT` to be consistent.

File: tests/sqlparser_common.rs [hunk 2]
```diff
@@ -1023,7 +1023,7 @@ fn parse_create_table() {
                constrained INT NULL CONSTRAINT pkey PRIMARY KEY NOT NULL UNIQUE CHECK (constrained > 0),
                ref INT REFERENCES othertable (a, b),\
                ref2 INT references othertable2 on delete cascade on update no action\
-               )";
+               ) WITHOUT ROWID";
```
[the remark this answers is not in the document]
Comment (mashuai) [reply tests/sqlparser_common.rs:1026]: That's good, I add a new `sqlparser_sqlite.rs` to test that.

File: tests/sqlparser_sqlite.rs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        Statement::CreateTable {
+            name,
+            columns,
+            constraints,
+            with_options: _with_options,
+            if_not_exists: false,
+            external: false,
+            file_format: None,
+            location: None,
+            query: _query,
```
Comment (Dandandan) [inline tests/sqlparser_sqlite.rs:37]: maybe also use `query: None` or keep `with_options` and `query` out of the test (`,..` as well)?

File: src/ast/mod.rs [hunk 4]
```diff
@@ -689,10 +689,8 @@ impl fmt::Display for Statement {
                 if let Some(query) = query {
                     write!(f, " AS {}", query)?;
                 }
-                if let Some(without_rowid) = *without_rowid {
-                    if without_rowid {
-                        write!(f, " WITHOUT ROWID")?;
-                    }
+                if *without_rowid {
+                    write!(f, " WITHOUT ROWID")?;
                 }
```
Comment (nickolay) [inline src/ast/mod.rs:694]: This ought to be moved as well, to mirror the changes in the parsing code.

**Comment by nickolay** [comment]
I also unresolved two comments that I don't think you have addressed.

File: tests/sqlparser_common.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (nickolay) [reply tests/sqlparser_common.rs:1026]: .. and don't add it to a test that runs for all dialects, I meant.

**Comment by nickolay** [comment]
Awesome, thanks!

**Pull request merged**